libr/debug/session.c

```diff
@@ -244,9 +251,24 @@ R_API void r_debug_session_save(RDebug *dbg, const char *file) {
 	RSnapEntry snapentry;
 
 	ut32 i;
+	char *base_file, *diff_file;
+	const char *path = dbg->snap_path;
+	if (!r_file_is_directory (path)) {
+		eprintf ("%s is not correct path\n", path);
+		return;
+	}
+	base_file = r_str_newf ("%s/%s.dump", path, file);
```

Comment:
define the variables here. char *base_file ... do not use different lines to define and assign. like it was before